packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,242 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    CarouselHandledProps,
+    CarouselProps,
+    CarouselUnhandledProps,
+    Slide,
+    SlideTheme,
+} from "./carousel.props";
+import { Flipper, FlipperDirection } from "../flipper";
+import { Tabs } from "@microsoft/fast-components-react-base";
+import { TabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * The carousel state interface
+ */
+export interface CarouselState {
+    /**
+     * Holds the active tab id to share with other controls
+     */
+    activeId: string;
+}
+
+class Carousel extends Foundation<
+    CarouselHandledProps,
+    CarouselUnhandledProps,
+    CarouselState
+> {
+    public static displayName: string = "Carousel";
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
+        nextProps: CarouselProps,
+        prevState: CarouselState
+    ): null | Partial<CarouselState> {
+        if (
+            typeof nextProps.activeId === "string" &&
+            nextProps.activeId !== prevState.activeId
+        ) {
+            return {
+                activeId: nextProps.activeId,
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<CarouselHandledProps> = {
+        managedClasses: void 0,
+        label: void 0,
+        activeId: void 0,
+        items: void 0,
+    };
+
+    private slides: Slide[];
+
+    /**
+     * Define constructor
+     */
+    constructor(props: CarouselProps) {
+        super(props);
+
+        if (Array.isArray(this.props.items)) {
+            this.state = {
+                activeId:
+                    typeof this.props.activeId === "string"
+                        ? this.props.activeId
+                        : get(this.props.items[0], "id", ""),
+            };
+
+            this.updateSlides();
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(): void {
+        if (this.slides !== this.props.items) {
```

Comment:
The way you have this set up, `this.slides` will never be strictly equal to `this.props.items` because `slice` returns a *new* array. I don't really think this is necessary though, if you make the changes I've discussed in the `updateSlides` method